File: src/main/java/no/nav/familie/ks/oppslag/dokarkiv/metadata/KontanstøtteSøknadMetadata.java
package no.nav.familie.ks.oppslag.dokarkiv.metadata;


public class KontanstøtteSøknadMetadata extends AbstractDokumentMetadata {
    public static final String TEMA = "KON";
    public static final String BEHANDLINGSTEMA = "ab0084"; // https://confluence.adeo.no/display/BOA/Behandlingstema
    public static final String KANAL = "NAV_NO";
    public static final String DOKUMENT_TYPE_ID = "KONTANTSTØTTE_SØKNAD";
    public static final String DOKUMENT_TITTEL = "Søknad om kontantstøtte til småbarnsforeldre";
    public static final String BREVKODE = "NAV 34-00.08";
    public static final String DOKUMENT_KATEGORI = "SOK";

    @Override
    public String getTema() {
        return TEMA;
    }

    @Override
    public String getBehandlingstema() {
        return BEHANDLINGSTEMA;
    }

    @Override
    public String getKanal() {
        return KANAL;
    }

    @Override
    public String getDokumentTypeId() {
        return DOKUMENT_TYPE_ID;
    }

    @Override
    public String getTittel() {
        return DOKUMENT_TITTEL;
    }

    @Override
    public String getBrevkode() {
        return BREVKODE;
    }

    @Override
    public String getDokumentKategori() {
        return DOKUMENT_KATEGORI;
    }
}

File: src/main/java/no/nav/familie/integrasjoner/dokarkiv/metadata/KontanstøtteSøknadMetadata.kt
package no.nav.familie.integrasjoner.dokarkiv.metadata

import no.nav.familie.integrasjoner.dokarkiv.client.domene.JournalpostType
import no.nav.familie.kontrakter.felles.Behandlingstema
import no.nav.familie.kontrakter.felles.Fagsystem
import no.nav.familie.kontrakter.felles.Tema
import no.nav.familie.kontrakter.felles.dokarkiv.Dokumenttype
import org.springframework.stereotype.Component

@Component
object KontanstøtteSøknadMetadata : Dokumentmetadata {
    override val journalpostType: JournalpostType = JournalpostType.INNGAAENDE
    override val fagsakSystem: Fagsystem? = null
    override val tema: Tema = Tema.KON
    override val behandlingstema: Behandlingstema = Behandlingstema.Kontantstøtte
    override val kanal: String = "NAV_NO"
    override val dokumenttype: Dokumenttype = Dokumenttype.KONTANTSTØTTE_SØKNAD
    override val tittel: String = "Søknad om kontantstøtte til småbarnsforeldre"
    override val brevkode: String = "NAV 34-00.08"
    override val dokumentKategori: Dokumentkategori = Dokumentkategori.SOK
}
